### PR TITLE
Updated to 1.8

### PR DESCRIPTION
**Pull request by Xyphos**
Updated to 1.8
Updated to .NET 4.0 Framework; KSP no longer supports .NET 3.5
compiled ok with no code changes, just review and change the reference paths in the project files.

### CODE REVIEW

**Comment by PiezPiedPy** [comment]
@Xyphos ty, I'll have a look over the weekend. 😉

File: src/Plugin/3rdParty/GLUtils.cs [hunk 1]
```diff
@@ -70,18 +72,37 @@ public static void DrawGroundMarker(CelestialBody body, double latitude, double
...
         }
 
-        public static void GLTriangle(Vector3d worldVertices1, Vector3d worldVertices2, Vector3d worldVertices3, Color c, bool map)
+        public static void GLTriangle(
+                Vector3d worldVertices1,
+                Vector3d worldVertices2,
+                Vector3d worldVertices3,
+                Color    c,
+                bool     map
+            )
```
Comment (PiezPiedPy) [inline src/Plugin/3rdParty/GLUtils.cs]: Please keep to original formatting 👍

File: src/Plugin/3rdParty/GLUtils.cs [hunk 2]
```diff
@@ -129,29 +150,55 @@ public static bool IsOccluded(Vector3d worldPosition,  CelestialBody byBody, Vec
...
+
+                int step = (dashed ? 2 : 1);
+                for (int i = 0; i < points.Count - 1; i += step)
                 {
-                    GLPixelLine(points[i], points[i + 1], map);
+                    if (!IsOccluded(
+                                points[i],
+                                mainBody,
+                                camPos
+                            )
```
Comment (PiezPiedPy) [inline src/Plugin/3rdParty/GLUtils.cs]: Same here Re-code styles, keep parameters of methods properties etc on a single line unless it exceeds far beyond the right margin.

File: src/Bootstrap/Bootstrap.csproj [hunk 3]
```diff
@@ -62,17 +63,26 @@
       <HintPath Condition="Exists('$(KSPDEVDIR)\KSP_x64_Data\Managed\Assembly-CSharp.dll')">$(KSPDEVDIR)\KSP_x64_Data\Managed\Assembly-CSharp.dll</HintPath>
       <HintPath Condition="Exists('..\DLLs\Assembly-CSharp.dll')">..\DLLs\Assembly-CSharp.dll</HintPath>
       <Private>False</Private>
+      <HintPath>..\..\..\..\..\..\..\Program Files (x86)\Steam\steamapps\common\Kerbal Space Program\KSP_x64_Data\Managed\Assembly-CSharp.dll</HintPath>
```
Comment (PiezPiedPy) [inline src/Bootstrap/Bootstrap.csproj]: You should be using the env vars to set the install paths

**Comment by PiezPiedPy** [comment]
I suggest you to have a look at the CONTRIBUTING.md file. The older versions of KSP use different dll's, so in order to build the release zip you need KSP1.3.1, 1.7.2 and the latest KSP release available installed in their own directories and set the env vars to point to the dev installs.

Probably be wise for you to update your clone/fork and submit a new PR with your GLUtils changes.

**Comment by Xyphos** [comment]
I use resharper, it auto-formats for me in the style I prefer; I hate long lines.
if my code style bothers you that much, I'll refrain from making future pull requests.

**Comment by PiezPiedPy** [comment]
> I use resharper, it auto-formats for me in the style I prefer; I hate long lines.
> if my code style bothers you that much, I'll refrain from making future pull requests.

If you feel it necessary to refrain yourself from making PR's due to having to make a few formatting changes then feel free to do so 🤦‍♂.

**Pull request closed without merging**